### ISSUE TITLE
[enhancement] Pass window id to exec in assignments

### ISSUE THREAD

**Issue by Airblader** [i3/i3#1729]
When calling external program in an assignment

```
for_window[class="(?i)firefox"] exec ~/scripts/do_some_stuff.sh
```

it would be helpful for the script to have information about the matched window so that it could make requests on i3's IPC for that window.

I propose setting an environment variable such as `$I3_MATCHED_WINDOW` to the window id so that the script could do a call like `i3-msg "[id=$I3_MATCHED_WINDOW] <command>"`.

Note: if #1697 gets merged, I'd like the equivalent for the clicked workspace button in those commands so that the custom script actually knows which button was clicked.

**Comment by acrisci** [comment]
I don't see how this is useful since the window is already matched and you can already run commands on it in the `for_window` command.

**Comment by Airblader** [comment]
Yes, if all you want to do is run a command, that is correct. However, think about only conditionally executing a command. The best way right now would be to duplicate the command criteria in your script:

````
if <something>:
    exec("i3-msg [class=(?i)firefox] move right")
```

And on top of that if multiple windows are matched, each match would execute the script, each run of which would then again execute the same command on *all* windows, which likely to mess things up.

**Comment by acrisci** [comment]
I think it would be a lot cleaner to do that within the ipc window event.

**Comment by Airblader** [comment]
Yeah, I guess you're right. Oh well, the idea for the workspace buttons lives on, though, but we'll cross that bridge when the PR is merged.
